x/nexus/keeper/rate_limit.go

```diff
@@ -42,7 +42,8 @@ func (k Keeper) RateLimitTransfer(ctx sdk.Context, chain exported.ChainName, ass
 	return nil
 }
 
-// SetRateLimit sets a rate limit for the given chain and asset
+// SetRateLimit sets a rate limit for the given chain and asset.
+// If max uint256 is provided as a limit, it's treated as a rate limit being infinite/not being set.
 func (k Keeper) SetRateLimit(ctx sdk.Context, chainName exported.ChainName, limit sdk.Coin, window time.Duration) error {
```

Comment:
why do we need to pack the deletion rule in with the setter? Also, this seems to break the invariant of always having a rate limit once it's set, that you fought so hard for

Reply:
i'll create another refactor PR based on our discussion, and we can merge it if we have time for this release